### ISSUE TITLE
cmake won't work on Ubuntu EGLIBC 2.19-0ubuntu6.13 (the Travis CI linux image)

### ISSUE THREAD

**Issue by maelvls** [Linuxbrew/brew#830]
**Please fill out the issue checklist below and provide all the requested information.**

- [x] ran a `brew` command and reproduced the problem with multiple formulae? 
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?
<details><pre>
travis@a09c25487980:/$ brew config
HOMEBREW_VERSION: 1.7.1
ORIGIN: https://github.com/Linuxbrew/brew
HEAD: 468d869022ebe725d7b7745adfa872db3e6c16c9
Last commit: 4 weeks ago
Core tap ORIGIN: https://github.com/Linuxbrew/homebrew-core
Core tap HEAD: 305f8f42fb9f5890d6e3a88372c27cec7f0a72d8
Core tap last commit: 10 hours ago
HOMEBREW_PREFIX: /home/linuxbrew/.linuxbrew
HOMEBREW_REPOSITORY: /home/linuxbrew/.linuxbrew/Homebrew
HOMEBREW_CELLAR: /home/linuxbrew/.linuxbrew/Cellar
HOMEBREW_CACHE: /home/travis/.cache/Homebrew
HOMEBREW_BOTTLE_DEFAULT_DOMAIN_LINUX: https://linuxbrew.bintray.com
HOMEBREW_BOTTLE_DEFAULT_DOMAIN_MACOS: https://homebrew.bintray.com
CPU: quad-core 64-bit kabylake
Homebrew Ruby: 2.3.7 => /home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.7/bin/ruby
Clang: N/A
Git: 2.15.1 => /usr/bin/git
Curl: 7.35.0 => /usr/bin/curl
Java: N/A
Kernel: Linux 4.9.93-linuxkit-aufs x86_64 GNU/Linux
OS: Ubuntu 14.04.5 LTS (trusty)
Host glibc: 2.19
/usr/bin/gcc: 4.8.4
glibc: N/A
gcc: N/A
xorg: N/A
</pre></details>

- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
<details>
<pre>
travis@a09c25487980:/$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and what additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew provided
script of the same name. We found the following "config" scripts:
  /home/travis/.phpenv/shims/php-config
</pre>
</details>

To help us debug your issue please explain:
- What you were trying to do (and why)

I was trying to use Travis CI to build linux bottles for [touist/homebrew-touist](https://github.com/touist/homebrew-touist). 

- What happened (include command output)

The build fails with errors on Glib ([here](https://travis-ci.org/touist/homebrew-touist/jobs/421065832) is the full log of a faulty travis build): 
```
cmake: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by cmake)
cmake: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by cmake)
cmake: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by cmake)
```
## Step-by-step reproduction instructions

In order to reproduce using exactly the same env, I use docker for launching a local travis instance:

    docker run -it travisci/ci-garnet:packer-1512502276-986baf0 bash
    su travis
    yes | sh -c "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install.sh)"
    export PATH="/home/linuxbrew/.linuxbrew/bin:$PATH"
    brew install cmake
    cmake

Some more information on this travis image:

-  glib 2.19 (`ldd --version` gives `Ubuntu EGLIBC 2.19-0ubuntu6.13`)
- gcc 4.8.4 (`gcc --version`  gives `Ubuntu 4.8.4-2ubuntu1~14.04.3`)

## What I understand from this error

I guess this is related to the "new C++ ABI" thing. In a recent commit (400af242fd7252edc461), it is mentioned that glibc 2.19 is required. Do you think libstdc++.so.6 should also be >= 0.20? In fact:

    # travis@a09c25487980:/$ ls -al /usr/lib/x86_64-linux-gnu/libstdc++.so.6
    lrwxrwxrwx 1 root root 19 May  7  2016 /usr/lib/x86_64-linux-gnu/libstdc++.so.6 -> libstdc++.so.6.0.19

From the errors above, I deduce that the minimum runtime requirements for using `cmake` are libstdc++.so.6.0.21 at least (from this [table](https://github.com/FoldingAtHome/fah-issues/issues/1147)). Although the travis image has glib 2.19, it won't work.

Ideas?

Note: I posted in linuxbrew/brew instead of linuxbrew/homebrew-core because I suspect that this issue is related to `brew`, not the formulas.

Again, thank you for your awesome work!

**Comment by maelvls** [comment]
Hmm... I forgot to mention the most important information: the git revisions of Homebrew-core and brew

- **Brew**: version 1.7.1, commit 468d869 (July 28 2018) 
  with `git -C /home/linuxbrew/.linuxbrew/Homebrew log -n1`
- **Homebrew-core**: 0953321e (Aug 26 2018)
  with `git -C /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core log -n1`


Then I realized that 1.7.1 was released one month ago. So I tried switching to the master version (git pull in the Homebrew folder) and tadaa! It works perfectly now!

So the next release (1.7.2 or so) will definitely fix my problem 😀
